_clients/ruby.md

```diff
@@ -0,0 +1,528 @@
+    request_timeout: 120,
+    log: true
+  )
+
+client.cluster.health
+```
+
+The output is as follows:
+
+```bash
+2022-08-25 14:24:52 -0400: GET http://localhost:9200/ [status:200, request:0.048s, query:n/a]
+2022-08-25 14:24:52 -0400: < {
+  "name" : "opensearch",
+  "cluster_name" : "docker-cluster",
+  "cluster_uuid" : "Aw0F5Pt9QF6XO9vXQHIs_w",
+  "version" : {
+    "distribution" : "opensearch",
+    "number" : "2.2.0",
+    "build_type" : "tar",
+    "build_hash" : "b1017fa3b9a1c781d4f34ecee411e0cdf930a515",
+    "build_date" : "2022-08-09T02:27:25.256769336Z",
+    "build_snapshot" : false,
+    "lucene_version" : "9.3.0",
+    "minimum_wire_compatibility_version" : "7.10.0",
+    "minimum_index_compatibility_version" : "7.0.0"
+  },
+  "tagline" : "The OpenSearch Project: https://opensearch.org/"
+}
+
+2022-08-25 14:24:52 -0400: GET http://localhost:9200/_cluster/health [status:200, request:0.018s, query:n/a]
+2022-08-25 14:24:52 -0400: < {"cluster_name":"docker-cluster","status":"yellow","timed_out":false,"number_of_nodes":1,"number_of_data_nodes":1,"discovered_master":true,"discovered_cluster_manager":true,"active_primary_shards":10,"active_shards":10,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":8,"delayed_unassigned_shards":0,"number_of_pending_tasks":0,"number_of_in_flight_fetch":0,"task_max_waiting_in_queue_millis":0,"active_shards_percent_as_number":55.55555555555556}
+```
+
+## Creating an index 
+
+You don't need to create an index explicitly in OpenSearch. Once you upload a document into an index that does not exist, OpenSearch creates the index automatically. Alternatively, you can create an index explicitly to specify settings like the number of primary and replica shards. To create an index with non-default settings, create an index body hash with those settings:
+
+```ruby
+index_body = {
+    'settings': {
+        'index': {
+        'number_of_shards': 1,
+        'number_of_replicas': 2 
+        }
+    }
+} 
+
+client.indices.create(
+    index: 'students',
+    body: index_body
+)
+```
+
+## Mappings
+
+OpenSearch uses dynamic mapping to infer field types of the documents that are indexed. However, to have more control over the schema of your document, you can pass an explicit mapping to OpenSearch. You can define data types for some or all fields of your document in this mapping. To create a mapping for an index, use the `put_mapping` method:
+
+```ruby
+client.indices.put_mapping(
+    index: 'students', 
+    body: {
+        properties: {
+            first_name: { type: 'keyword' },
+            last_name: { type: 'keyword' }
+        }  
+    }
+)
+```
+
+By default, string fields are mapped as `text`, but in the mapping above both `first_name` and `last_name` fields are mapped as `keyword`. This mapping signals to OpenSearch that this field should not be analyzed and should support only full case-sensitive matches.
+
+You can verify the index's mappings using the `get_mapping` method:
```

Comment:
Can "mappings" be "mapping" here?

Reply:
I like "mappings" because there are multiple. I agree that the method name does not agree with this, but in the REST API we have both methods "get mapping" and "get mappings" which do exactly the same thing, so the users will be used to the nomenclature.